File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        unique = set()
        repeating = set()
        for c in s:
            if c in repeating:
                continue
            if c in unique:
                repeating.add(c)
                unique.remove(c)
            else:
                unique.add(c)
        for index in range(len(s)):
            if s[index]  in unique:
                return index
        return -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_set<char> unique;
        unordered_set<char> repeating;
        for(int i = 0; i < s.size() ; i++){
            if(repeating.count(s[i]))
                continue;
            else if(unique.count(s[i])){
                unique.erase(s[i]);
                repeating.insert(s[i]);
            }else{
                unique.insert(s[i]);
            }
        }
        for(int i = 0 ; i < s.size() ; i++){
            if(unique.count(s[i]))
                return i;
        }
        return -1;
    }
};